enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,600 @@
+[Effectuating Subnet Updates](#effectuating-subnet-updates) will mitigate
+cluster admins from experiencing this issue.
+
+### Drawbacks
+
+Requiring the cluster admin to delete the LoadBalancer-type
+service leads to several minutes of ingress traffic disruption.
+
+This enhancement brings additional engineering complexity for upgrade
+scenarios because cluster admins have previously been allowed to directly
+add this annotation on a service. 
+
+Debugging invalid subnets will be confusing for cluster admins and may
+lead to extra support cases or bugs. By requiring service to be recreated,
+it guarantees that the service won't be provisioned when invalid subnets
+are provided, thus clearly indicating to the cluster admin that something
+is wrong.
+
+The design in [Effectuating Subnet Updates](#effectuating-subnet-updates)
+requires a cluster admin to check the IngressController's status after
+updating `Subnets` for instructions on how to proceed.
+
+## Open Questions
+
+- Do we need to configure subnets for default IngressController at cluster installation (i.e. Day 1 API)?
+  - **Answer**: Yes. Service Delivery needs the ability to pick subnets for the default IngressController during
+    installation for ROSA. However, this work is being done as a separate epic (https://issues.redhat.com/browse/CORS-3440),
+    which includes redesigning the installconfig fields and deprecating the current `platform.aws.subnets` field.
+    We will address an install-time API in a separate enhancement proposal.
+- Should we create an alert for when the IngressController is in a `Progressing` status due to the inconsistency
+  between subnets?
+
+## Test Plan
+
+Unit tests as well as E2E tests will be added to the Ingress Operator
+repository.
+
+E2E tests will cover the following scenarios:
+
+- Creating an IngressController with `Subnets` and observing
+  `service.beta.kubernetes.io/aws-load-balancer-subnets` on the LoadBalancer-type Service.
+- Updating an IngressController with new `Subnets`, deleting the
+  LoadBalancer-type service, and observing `service.beta.kubernetes.io/aws-load-balancer-subnets`
+  (as described in [Updating an existing IngressController with new Subnets Workflow](#updating-an-existing-ingresscontroller-with-new-subnets-workflow)).
+- Directly configuring `service.beta.kubernetes.io/aws-load-balancer-subnets` on the
+  LoadBalancer-type service and setting `Subnets` on the IngressController while observing
+  `LoadBalancerProgressing` transitioning back to `Status: False` (as described in
+  [Unmanaged Subnet Annotation Migration Workflow](#unmanaged-subnet-annotation-migration-workflow)).
+- Creating a IngressController with the `ingress.operator.openshift.io/auto-delete-load-balancer`
+  annotation, updating an IngressController with new `Subnets`, and observing the service
+  get automatically deleted and recreated with `service.beta.kubernetes.io/aws-load-balancer-subnets`
+  configured to the new `Subnets`.
```

Comment:
It would be interesting to see what happens when you haven't used the auto-delete annotation, but then add it afterward.  Does it do the deletion if you haven't added the annotation ahead of time?

Reply:
Yup just tried this out in a real cluster to confirm.

If you add the auto-delete annotation after updating the subnets, it will immediately delete the service when you add the auto-delete annotation:

1. Updating the IngressController's annotation triggeres a reconcile.
2. The reconcile will run the [ensureLoadBalancerService](https://github.com/openshift/cluster-ingress-operator/blob/13a29ff4029823bbd4a06924843fbe4cba9fba0a/pkg/operator/controller/ingress/controller.go#L1078) function
3. The [updateLoadBalancerService](https://github.com/openshift/cluster-ingress-operator/blob/13a29ff4029823bbd4a06924843fbe4cba9fba0a/pkg/operator/controller/ingress/load_balancer_service.go#L581) will now see the auto-delete annotation as well as the need to recreate the LB and it will do so immediately. 

This was how Miciah designed in when implementing https://github.com/openshift/enhancements/blob/master/enhancements/ingress/mutable-publishing-scope.md.